### ISSUE TITLE
Add support for lazy reload during schema evolution

### ISSUE THREAD

**Issue by deemoliu** [apache/pinot#14499]
Currently schema updates and schema reload are not atomic operations. if we add new columns without schema reload, 
1. there is no force commit for the consuming segments
2. by default the new segment will ingest with new schema 
3. the old segment will not reflect the new column unless the reload is triggered.
4. when query the newly added column or `select *`, there will be merge exceptions on segments with different schema and produce query failure to customer. Customer need to perform reload or change query to bypass the error.

For some scenarios, we also schema evolution without reload on purpose. e.g. many tables with massive data and segments, it's  risky to perform schema reload safety because it might overwhelm heap. The inconsistent schema on Pinot segment and schema will be hard to maintain.  

We proposed the following approach to solve this.
Add an config (so that we don't change current behavior) and support Lazy reload - if in the query there are new columns doesn’t exist in one of the segment, just return default value for it

other options includes,
- add a query option (so that we don't change current behavior) and do not return result for the newly added column until completely reloaded. 
- support Schema versioning in Pinot.

**Comment by deemoliu** [comment]
@rohityadav1993 do you want to work on this feature?

**Comment by rohityadav1993** [comment]
@deemoliu, yes, will share a doc in coming days based on the suitable approach for this.
